packages/apex-node/src/utils/traceFlags.ts

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Connection } from '@salesforce/core';
+import * as util from 'util';
+import { nls } from '../i18n';
+import { DEFAULT_DEBUG_LEVEL_NAME, LOG_TYPE } from '../logs/constants';
+
+interface UserRecord {
+  Id: string;
+}
+
+interface DebugLevelRecord {
+  ApexCode: string;
+  VisualForce: string;
+}
+
+interface TraceFlagRecord {
+  Id: string;
+  LogType: string;
+  DebugLevelId: string;
+  StartDate: Date | undefined;
+  ExpirationDate: Date | undefined;
+  DebugLevel: DebugLevelRecord;
+}
+
+interface DataRecordResult {
+  id?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  errors?: any[];
+  success: boolean;
+}
+
+interface IdRecord {
+  Id: string;
+}
+
+interface QueryRecord {
+  totalSize: number;
+  records: IdRecord[];
+}
+
+export class TraceFlags {
+  private readonly LOG_TIMER_LENGTH_MINUTES = 30;
```

Comment:
What's the difference between the `TAIL_LISTEN_TIMEOUT_MIN` and `LOG_TIMER_LENGTH_MINUTES` variables here?

Reply:
I'm not sure there needs to be (LOG_TIMER_LENGTH_MINUTES was just carry over from the trace flag class in the extension). Do you have any thoughts on renaming it for clarity though?

Reply:
I decided I liked 'LOG_TIMER_LENGTH_MINUTES' best and removed TAIL_LISTEN_TIMEOUT_MIN, but I can change it if that doesn't feel obvious enough for both cases.